lib/segment/src/types.rs

```diff
@@ -45,7 +45,41 @@ pub type FloatPayloadType = f64;
 /// Type of integer point payload
 pub type IntPayloadType = i64;
 /// Type of datetime point payload
-pub type DateTimePayloadType = chrono::DateTime<chrono::Utc>;
+pub type DateTimePayloadType = DateTimeWrapper;
+
+/// Wraps `DateTime<Utc>` to allow more flexible deserialization
+#[derive(Clone, Copy, Serialize, JsonSchema, Debug, PartialEq, PartialOrd)]
+#[serde(transparent)]
+pub struct DateTimeWrapper(pub chrono::DateTime<chrono::Utc>);
```

Comment:
Maybe name it just as `DateTime`? And use it everywhere in the codebase instead of `chrono::DateTime`, `chrono::NaiveDateTime`, `std::time::SystemTime`, etc. This would improve the consistency.

Reply:
Not sure if we should replace `std::time::SystemTime`, but I thought about the name, and decided `DateTimeWrapper` would be a good name to make it more obvious when seeing this in inlay hints, or elsewhere outside the context of this PR. Do you find it weird?

Reply:
Since this name is not exposed to any API, I am ok to merge as is and change later if it causes any confusion